sync/block_listener.go

```diff
@@ -67,9 +67,9 @@ func (bl *BlockListener) ListenToGossipBlocks() {
 					bl.Error("got empty message while listening to gossip blocks")
 					return
 				}
-
+				tmr := newMilliTimer(gossipBlockTime)
```

Comment:
creating a new timer for each block is a bad practice which will eventually blow up our memory/metrics server.
this should be done with an histogram/summary

Reply:
ok, my bad, looking at the impl I see you use the same histogram just create a new timer each time.